bert/model.py

```diff
@@ -0,0 +1,246 @@
+from typing import Optional
+from dataclasses import dataclass
+from mlx.utils import tree_unflatten, tree_map
+from mlx.nn.layers.base import Module
+from mlx.nn.layers.linear import Linear
+from mlx.nn.layers.normalization import LayerNorm
+from transformers import AutoTokenizer
+
+import mlx.core as mx
+import mlx.nn as nn
+import argparse
+import numpy
+import math
+
+
+@dataclass
+class ModelArgs:
+    intermediate_size: int = 768
+    num_attention_heads: int = 12
+    num_hidden_layers: int = 12
+    vocab_size: int = 30522
+    attention_probs_dropout_prob: float = 0.1
+    hidden_dropout_prob: float = 0.1
+    layer_norm_eps: float = 1e-12
+    max_position_embeddings: int = 512
+
+
+model_configs = {
+    "bert-base-uncased": ModelArgs(),
+    "bert-base-cased": ModelArgs(),
+    "bert-large-uncased": ModelArgs(
+        intermediate_size=1024, num_attention_heads=16, num_hidden_layers=24
+    ),
+    "bert-large-cased": ModelArgs(
+        intermediate_size=1024, num_attention_heads=16, num_hidden_layers=24
+    ),
+}
+
+
+class MultiHeadAttention(Module):
+    """
+    Minor update to the MultiHeadAttention module to ensure that the
+    projections use bias.
+    """
```

Comment:
Is the `bias` really the only change from the `nn` version? If so we should just add an option to that version.. would greatly reduce duplication!

Reply:
I added an additional helper function to the class to convert the HuggingFace tokenizer's {0, 1} mask to an additive causal mask, but that doesn't need to exist **on** the `MultiHeadAttention` class. So, yes, the existing MLX multi-head attention implementation could simply take a `use_bias` param and I could remove it from this implementation. I can test it out and submit a PR to both placed if it works.

Reply:
Perfect! Though I would use the name `bias` rather than `use_bias` for consistency with our `Linear` layer etc.